restord/ImageSessionServer.m

```diff
@@ -110,34 +110,61 @@ - (void)beginImaging {
   int asrReturnCode = [self applyImage];
   NSLog(@"%@ ASR exit code: %d", self, asrReturnCode);
 
-  NSError *err;
-  if (asrReturnCode == 0) {
-    DADiskRef disk = NULL;
-
-    // Get the synthesized "Macintosh HD" apfs disk, if any.
-    NSString *apfsOSDisk = [self apfsOSDisk];
-    if (apfsOSDisk) {
-      NSString *apfsBootDisk = [NSString stringWithFormat:@"/dev/%@", apfsOSDisk];
-      disk = DADiskCreateFromBSDName(NULL, self.diskArbSession, apfsBootDisk.UTF8String);
-    }
-
-    NSURL *mountURL = [self mountDisk:disk ?: self.diskRef];
-    if (mountURL) {
-      NSLog(@"%@ Remounted target %@ as %@", self, self.destination.path, mountURL.path);
-      [self applyImageInfo:mountURL];
-      [self blessMountURL:mountURL];
-      [self unmountDisk:disk ?: self.diskRef];
-    } else {
-      NSLog(@"%@ Unable to remount target, skipping imaginfo.plist application", self);
-    }
+  if (asrReturnCode != 0) {
+    NSError *err = [self constructErrorFromReturnCode:asrReturnCode];
+    [[self.client remoteObjectProxy] imageAppliedSuccess:NO error:err];
+    return;
+  }
 
+  // Get the synthesized "Macintosh HD" apfs disk, if any.
+  DADiskRef disk = NULL;
+  NSString *apfsOSDisk = [self apfsOSDisk];
+  if (apfsOSDisk) {
+    NSString *apfsBootDisk = [NSString stringWithFormat:@"/dev/%@", apfsOSDisk];
+    disk = DADiskCreateFromBSDName(NULL, self.diskArbSession, apfsBootDisk.UTF8String);
+  }
+  NSURL *mountURL = [self mountDisk:disk ?: self.diskRef];
+
+  // Fail if the disk would not mount and the post script is required.
+  // Succeed if the disk would not mount and the post script is not required.
+  // Otherwise continue on.
+  if (!mountURL && self.image.postScript && self.image.postScriptMustSucceed) {
```

Comment:
suggestion: use 
```
if (!mountURL) { 
  if (self.image.postScript && self.image.postScriptMustSucceed) {
  } else {
  }
}
```
and share the last two lines.

Reply:
Nice, worth the nest.